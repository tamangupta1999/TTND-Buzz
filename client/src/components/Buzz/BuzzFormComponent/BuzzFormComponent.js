import React, { useState } from 'react';
import Styles from './BuzzFormComponent.module.css';
import { toast } from 'react-toastify';

const BuzzFormComponent = (props) => {
    const { email, buzzPostHandler, uiCategory } = props;
    const [buzzMessage, setBuzzMessage] = useState('');
    const [category, setcategory] = useState('');
    const [buzzImage, setBuzzImage] = useState([]);

    const validateForm = () => {
        if (buzzMessage.length === 0) {
            toast.warning('Buzz Message Is Required');
            return true;
        } else if (category.length === 0) {
            toast.warning('Category Is Required');
            return true;
        }
        else {
            return false;
        }
    }

    const formSubmit = (event) => {
        event.preventDefault();
        let isValid = validateForm();
        if (!isValid) {
            const formdata = new FormData();
            formdata.append("buzzMessage", buzzMessage);
            formdata.append("category", category);
            for (const image of buzzImage) {
                formdata.append("buzzImage", image);
            }
            formdata.append("userId", email);
            buzzPostHandler(formdata);
            setBuzzMessage('');
            setcategory('');
            setBuzzImage('');
        }
    }
    const fileChange = async (event) => {
        await setBuzzImage(event.target.files);
    }
    return (
        <div className={Styles.BuzzFormComponent}>
            <h3 className={Styles.BuzzHeader}><i className="fas fa-edit"></i> Create Buzz</h3>
            <form onSubmit={formSubmit} encType="multipart/form-data">
                <textarea data-testid="thoughts" className={Styles.TextArea} placeholder="Share Your Thoughts......" value={buzzMessage} onChange={event => setBuzzMessage(event.target.value)}></textarea>
                <div className={Styles.FormBottom}>
                    <div>
                        <select data-testid="category" className={Styles.SelectTag} name='Category' value={category} onChange={event => setcategory(event.target.value)}>
                            <option hidden>Category</option>
                            {uiCategory.map(categ => {
                                return <option key={categ} value={categ}>{categ}</option>
                            })}
                        </select>
                        <label htmlFor="file" className={Styles.ImageLabel}><i className="fas fa-image"></i>({buzzImage.length})</label>
                        <input type='file' id="file" name='buzzImage' accept="image/*"
                            style={{ visibility: 'hidden' }}
                            onChange={event => fileChange(event)} multiple />
                    </div>
                    <button data-testid="submit" className={Styles.SubmitBtn} type="submit"><i className="fas fa-angle-right fa-2x"></i></button>
                </div>
            </form>
        </div>
    )
}

export default BuzzFormComponent;
