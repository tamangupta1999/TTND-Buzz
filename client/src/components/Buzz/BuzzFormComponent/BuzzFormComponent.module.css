.BuzzFormComponent {
    margin: 0rem 3rem;
    margin-top: 0.9rem;
    width: 90%;
    border: 1px solid #ccc;
    background: #fff;
    box-shadow: 2px 3px 4px #ccc;
}
.BuzzHeader {
    margin: 0rem;
    padding: 0.5rem;
    border-bottom: 1px solid #ccc;
    background-color: #f4f4f4;
    font-size: 1rem;
    font-weight: bolder;
}
.BuzzHeader i {
    color: grey;
}
.TextArea {
    width:  99.5%;
    height: 6rem;
    border: none;
    outline: none;
    border-bottom: 1px solid #ccc;
}
.TextArea:focus{
    border: thin solid cornflowerblue;
}
.FormBottom {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem;
}
.SelectTag {
    border: none;
    cursor: pointer;
    background: none;
    outline: none;
    width: 5rem;
    margin-right: 2rem;
}
.ImageLabel {
    cursor: pointer;
    color: grey;
}

.SubmitBtn {
    border: none;
    background: none;
    color: orangered;
    cursor: pointer;
}
.Dark {
    color: black;
}

@media(max-width:768px){
    .BuzzFormComponent{
        width: 100%;
        margin: 0rem;
        margin-top: 1rem;
    }
}