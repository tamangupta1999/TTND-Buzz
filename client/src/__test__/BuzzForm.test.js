import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { cleanup, fireEvent, render } from '@testing-library/react';

import BuzzFormComponent from '../components/Buzz/BuzzFormComponent/BuzzFormComponent';


const defaultProps = {
    email: 'dummy@email.com',
    buzzPostHandler: jest.fn(() => console.log('value')),
    uiCategory: ['ACTIVITY', 'LOST_AND_FOUND']
}

afterEach(cleanup);

describe('Rendering <BuzzFormComponent />', () => {
    // it('should Component render ', () => {
    //     const wrapper = render(<BuzzFormComponent {...defaultProps} />);
    //     expect(wrapper).toMatchSnapshot();
    // });

    it('verifying Category with default props', () => {
        const wrapper = render(<BuzzFormComponent {...defaultProps} />).getByTestId('category');
        expect(wrapper.querySelectorAll('option').length).toEqual(3);
    });

    it('check button enable', () => {
        const { getByTestId } = render(<BuzzFormComponent {...defaultProps} />);
        expect(getByTestId('submit')).not.toBeDisabled();
    });

    it('should submit Button is working', () => {
        const { getByTestId } = render(<BuzzFormComponent {...defaultProps} />);
        let val = fireEvent.click(getByTestId('submit'))
        expect(val).toBe(true)
    });
})