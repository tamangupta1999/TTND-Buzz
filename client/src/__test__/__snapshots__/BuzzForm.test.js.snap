// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering <BuzzFormComponent /> should Component render  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="BuzzFormComponent"
      >
        <h3
          class="BuzzHeader"
        >
          <i
            class="fas fa-edit"
          />
           Create Buzz
        </h3>
        <form
          enctype="multipart/form-data"
        >
          <textarea
            class="TextArea"
            data-testid="thoughts"
            placeholder="Share Your Thoughts......"
          />
          <div
            class="FormBottom"
          >
            <div>
              <select
                class="SelectTag"
                data-testid="category"
                name="Category"
              >
                <option
                  hidden=""
                >
                  Category
                </option>
                <option
                  value="ACTIVITY"
                >
                  ACTIVITY
                </option>
                <option
                  value="LOST_AND_FOUND"
                >
                  LOST_AND_FOUND
                </option>
              </select>
              <label
                class="ImageLabel"
                for="file"
              >
                <i
                  class="fas fa-image"
                />
                (
                0
                )
              </label>
              <input
                accept="image/*"
                id="file"
                multiple=""
                name="buzzImage"
                style="visibility: hidden;"
                type="file"
              />
            </div>
            <button
              class="SubmitBtn"
              data-testid="submit"
              type="submit"
            >
              <i
                class="fas fa-angle-right fa-2x"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="BuzzFormComponent"
    >
      <h3
        class="BuzzHeader"
      >
        <i
          class="fas fa-edit"
        />
         Create Buzz
      </h3>
      <form
        enctype="multipart/form-data"
      >
        <textarea
          class="TextArea"
          data-testid="thoughts"
          placeholder="Share Your Thoughts......"
        />
        <div
          class="FormBottom"
        >
          <div>
            <select
              class="SelectTag"
              data-testid="category"
              name="Category"
            >
              <option
                hidden=""
              >
                Category
              </option>
              <option
                value="ACTIVITY"
              >
                ACTIVITY
              </option>
              <option
                value="LOST_AND_FOUND"
              >
                LOST_AND_FOUND
              </option>
            </select>
            <label
              class="ImageLabel"
              for="file"
            >
              <i
                class="fas fa-image"
              />
              (
              0
              )
            </label>
            <input
              accept="image/*"
              id="file"
              multiple=""
              name="buzzImage"
              style="visibility: hidden;"
              type="file"
            />
          </div>
          <button
            class="SubmitBtn"
            data-testid="submit"
            type="submit"
          >
            <i
              class="fas fa-angle-right fa-2x"
            />
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
